packages/actor-init-sparql/test/HttpServiceSparqlEndpoint-test.ts

```diff
@@ -870,3 +871,76 @@ describe('HttpServiceSparqlEndpoint', () => {
     });
   });
 });
+
+const variantsDefault = [{ type: 'application/json', quality: 1 },
+  { type: 'simple', quality: 1 },
+  { type: 'application/sparql-results+json', quality: 1 },
+  { type: 'application/sparql-results+xml', quality: 1 },
+  { type: 'table', quality: 1 },
+  { type: 'tree', quality: 0.9 },
+  { type: 'stats', quality: 1 },
+  { type: 'application/trig', quality: 1 },
+  { type: 'application/n-quads', quality: 0.7 },
+  { type: 'text/turtle', quality: 0.6 },
+  { type: 'application/n-triples', quality: 0.3 },
+  { type: 'text/n3', quality: 0.2 },
+  { type: 'application/ld+json', quality: 0.9 }];
+
+describe('determineMediaType', () => {
+  let instance: any;
+  let request: any;
+  let response: any;
+  let variants: any;
+
+  beforeEach(async() => {
+    instance = new HttpServiceSparqlEndpoint({});
+    instance.writeQueryResult = jest.fn();
+    request = makeRequest();
+    response = new ServerResponseMock();
+  });
+
+  function makeRequest() {
+    request = stringToStream('default_test_request_content');
+    request.url = 'url_sparql';
+    request.headers = { 'content-type': 'contenttypewhichdefinitelydoesnotexist',
+      accept: 'application/json,application/trig,simple,stats' };
+    request.method = 'GET';
+    return request;
+  }
+
+  it('should return a null mediatype when given */* accept header'
+    , async() => {
+      request.url = 'url_undefined_query';
+      request.headers = { 'content-type': 'contenttypewhichdefinitelydoesnotexist',
+        accept: '*/*' };
+      expect(instance.determineMediaType(request, variants)).toEqual(null);
+    });
+
+  it('should return a null mediatype when no accept header'
+    , async() => {
+      request.url = 'url_undefined_query';
+      request.headers = { 'content-type': 'contenttypewhichdefinitelydoesnotexist' };
+      expect(instance.determineMediaType(request, variants)).toEqual(null);
+    });
+
+  it('should choose default mediatype if it is present in accept', async() => {
+    request.headers = { accept: 'application/json,application/trig,simple,stats' };
+    instance.parseBody = jest.fn(() => Promise.resolve('test_result'));
+    request.method = 'POST';
+    expect(instance.determineMediaType(request, variantsDefault)).toEqual('application/json');
+  });
+
+  it('should choose second preferred mediatype if it is present in accept', async() => {
+    request.headers = { accept: 'application/trig,simple,stats' };
+    instance.parseBody = jest.fn(() => Promise.resolve('test_result'));
+    request.method = 'POST';
+    expect(instance.determineMediaType(request, variantsDefault)).toEqual('simple');
+  });
+
+  it('should choose the next preferred variant available in accept', async() => {
+    request.headers = { accept: 'application/trig,stats' };
```

Comment:
Could you also add some test to check what happens with different q-values in the accept header?